internal/store/runtime_state.go

```diff
@@ -80,7 +86,7 @@ func (l LocalRuntimeState) RuntimeStatusError() error {
 }
 
 func (l LocalRuntimeState) HasEverBeenReadyOrSucceeded() bool {
-	return l.HasSucceededAtLeastOnce
+	return l.HasSucceededAtLeastOnce && !l.LastReadyTime.IsZero()
```

Comment:
It is unfortunate and confusing that the method name is "ready *or* succeeded" and this expression is "succeeded *and* ready"! I don't think this is buggy, but seems like a bit of a trap.

The "ready" in the method name means "service was started and is ready for traffic" while the "ready" in the expression means "is not waiting for the result of a readiness probe". I feel like the former is less likely to confuse the reader.

A couple of options that I haven't thought through too deeply:
1) move the block for if ReadinessProbe == nil -> LastReadyTime = time.Now to the [bit in Upper that sets HasSucceededAtLeastOnce](https://github.com/tilt-dev/tilt/blob/8a13e3876f4bcb7d02ff3d03463a83df0913f760/internal/engine/upper.go#L465) (or maybe make a RecordSuccess method takes care of both)
2) move the block for if ReadinessProbe == nil -> LastReadyTime = time.Now into local controller. i.e., when we'd create a prober, if there's no probe configured, just immediately emit a LocalServeReadinessProbeAction

I think (2) feels better to me but maybe there's some complication

Reply:
(also fine to clean this up in a later PR if you're coming back for integration tests)

Reply:
Okay - I think 6df97df2 represents a reasonable middle ground, which moves it closer to how the K8s runtime state works in that there's a single timestamp that drives everything. The logic to update that or not is at a higher-level so the state object is (slightly) more of a dummy data object.

A future, broader refactor might be in order across the runtime state objects + actions/reducers to more broadly move logic upwards, but hopefully it's not too confusing in this state.